haystack/components/embedders/types/protocol.py

```diff
@@ -0,0 +1,30 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, Dict, Protocol, TypeVar
+
+T = TypeVar("T", bound="TextEmbedder")
+
+# See https://github.com/pylint-dev/pylint/issues/9319.
+# pylint: disable=unnecessary-ellipsis
+
+
+class TextEmbedder(Protocol):
```

Comment:
👍 
Consistent with the ChatGenerator protocol, we don't use `@runtime_checkable` here because runtime checks would only verify method existence - not their signatures (see [mypy docs](https://mypy.readthedocs.io/en/stable/protocols.html#using-isinstance-with-protocols)) and this PR https://github.com/deepset-ai/haystack/pull/9037